src/workloads/execution/DensifyFillCombo.yml

```diff
@@ -0,0 +1,372 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: |
+  This workload tests the performance of queries that combine $densify and $fill stages.
+  $densify creates new documents to eliminate the gaps in the "timestamp" or "numeric"
+  fields. The other fields in these generates documents however, will be missing.
+  $fill set values for the output field (in the context of this workload,
+  "toFillRandomType" or "toFillNumeric") when the value is null or missing.
+
+GlobalDefaults:
+  dbname: &db test
+  batchSize: &batchSize 30000
+  fieldName: &field "timestamp"
+  index: &index
+    keys: {numeric: 1}
+    options: {name: "numeric"}
+  smallStepSize: &smallStep {^RandomInt: {min: 3, max: 3}}
+  nop: &Nop {Nop: true}
+
+Actors:
+- Name: CreateTimeSeriesCollection
+  Type: RunCommand
+  Threads: 1
+  Phases:
+  - Repeat: 1
+    Database: *db
+    Operation:
+      OperationMetricsName: CreateTimeSeriesCollection
+      OperationName: RunCommand
+      OperationCommand:
+        {create: &coll Collection0, timeseries: {timeField: "timestamp", metaField: "numeric"}}
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+
+- Name: InsertData
+  Type: Loader
+  Threads: 1
+  Phases:
+  - *Nop
+  - *Nop
+  - Repeat: 1
+    Database: *db
+    Threads: 1
+    CollectionCount: 1
+    DocumentCount: 100
```

Comment:
Because we are running hundreds of densify commands (we run each operation for 30 seconds), I put the initial document count very low and also increased the InternalQueryMaxAllowedDensifyDocs to 100k (default is 50k). I wonder if, for performance reasons, it would be better to change from duration based to repeat based @apodelko ?

Reply:
Could you please elaborate it a little further? I am not sure that I quite got what is going on here. If I understand correctly we have 8 different tests here and each has between 472 and 708 OperationsTotal (in the second patch) - so we have something like 5-6K OperationsTotal total.

Reply:
Sorry, I misspoke. We are running thousands not hundreds of densify queries (each operation is one query that contains a densify and fill stage). Without the `internalQueryMaxAllowedDensifyDocs` manually increased to 100k, we run into an error. `PlanExecutor error during aggregation :: caused by :: Generated 500001 documents in $densify, which is over the limit of 500000. Increase the 'internalQueryMaxAllowedDensifyDocs' parameter to allow more generated documents: generic server error`

Reply:
What in this case is a operation? As I see in the patch results, we have something like 5-6K OperationsTotal total for 8x30 sec runs. How many documents do we generate per operation?

Generically speaking, I'd rather suggest a larger database (so the number of generated document is rather small comparing with the initial size - as database size usually impact results a lot and we may have a slow down during the test). As per duration vs. number of iteration, no clear preference - but it is definitely better to run it for some time. Any idea how long it takes to run 10 times in this case? Probably shouldn't be less than 30 sec.

Reply:
I guess I was using test and operation synonymously since each [test](https://github.com/mongodb/genny/blob/PERF-2729/src/workloads/execution/DensifyFillCombo.yml#L225-L267) contains [one operation](https://github.com/mongodb/genny/blob/PERF-2729/src/workloads/execution/DensifyFillCombo.yml#L242-L263). But this was incorrect of me. 

I'm gonna get back to you on how many documents we generate per operation in these tests when I figure out how to measure that

Reply:
@apodelko I am looking at the csv I exported from genny. My understanding is that counters.n is documents created -- however each operation for every test has 0 counters.n? I could create a jstest for a similar database and run it and see from there how many docs are outputted?

Reply:
[DensifyTimestampFillWithoutPartition](https://docs.google.com/spreadsheets/d/1DrV8okba0Qa10M6JBQMMEr-I48ZzSkUUWbs8_6z8QVo/edit?usp=sharing) [DensifyTimestampFillValue](https://docs.google.com/spreadsheets/d/1mzW5in4yzo_Z-P9EYGlXr_T0fVztjrM14xzX7CPDZy4/edit#gid=917721369)  [DensifyNumericLinearFillNumeric](https://docs.google.com/spreadsheets/d/1AXyHpFV7TOu0M3IS4kkZ4DCtch1R3DcNPowrJtNpngI/edit?usp=sharing)  [DensifyNumericFillWithPartition](https://docs.google.com/spreadsheets/d/1bbbhTIWRoTjmmcPUKa94jucse_LMGNm_24oBpFBksrU/edit?usp=sharing)

Reply:
Is generating a large number of documents a problem? Would it make more sense to keep the densify limit at the default, and lower the ranges we're hitting? The limit was set as Bora felt that it was a good representation of the "average" users's largest operation, but it was very vague.

Reply:
@TedTuckman would that mean making explicit bounds, rather than just doing "full"?

Reply:
Either that or having fields with a smaller range for full. I don't feel that I have a good grasp on this discussion, so I leave it up to you guys

Reply:
@madelinezec I believe that we have each operation on a separate line in these csv files. So, for example, DensifyTimestampFillWithoutPartition has 62 lines - so we had 62 operations there. Apparently it is not a document per operations if we needed to increase  InternalQueryMaxAllowedDensifyDocs to 100K...

Reply:
Actually in your succesful patch it appears that we have 513 operations (lines) for DensifyTimestampFillWithoutPartition - which match the reported OperationThroughput metric of 17.12 ops/sec (and 30 sec duration).